internal/runners/config/set.go

```diff
@@ -0,0 +1,70 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/spf13/cast"
+)
+
+type Set struct {
+	out output.Outputer
+	cfg *config.Instance
+}
+
+type SetParams struct {
+	Key   string
+	Value string
+}
+
+func NewSet(prime primeable) *Set {
+	return &Set{prime.Output(), prime.Config()}
+}
+
+func (s *Set) Run(params SetParams) error {
+	err := validateKey(params.Key)
+	if err != nil {
+		return locale.WrapInputError(err, "err_config_invalid_key", "Invalid config key")
+	}
+
+	var value interface{}
+	value = params.Value
+	if v, ok := meta[strings.ToLower(params.Key)]; ok {
+		switch v.allowedType {
+		case Bool:
+			value = cast.ToBool(value)
+		case Int:
+			value = cast.ToInt(value)
+		}
+	}
+
+	err = s.cfg.Set(params.Key, value)
+	if err != nil {
+		return locale.WrapError(err, "err_cofing_set", fmt.Sprintf("Could not set value %s for key %s", value, params.Key))
+	}
+
+	err = setEvent(params.Key)
+	if err != nil {
+		logging.Error("Could not execute additional logic on config set")
+	}
+
+	s.out.Print(locale.Tl("config_set_success", "Successfully set config key: {{.V0}} to {{.V1}}", params.Key, params.Value))
+	return nil
+}
+
+func setEvent(key string) error {
+	value, ok := meta[key]
+	if !ok {
+		return nil
+	}
+
+	if value.setEvent == nil {
+		return nil
+	}
+
+	return value.setEvent()
+}
```

Comment:
nit: The key text casing is not handled when setEvent is called or within the function itself, but it is in the switch above. If that is a fundamental part of the meta map interaction, that map might be better made a custom map type that has methods to get values. Likewise for setting values. So, `meta.Set(key, getFn, nil)` where the nil is automatically set as `func() error { return nil} }` so that callers do not have to check nil, and `val, ok := meta.Get(key)` - both Get and Set methods would process the key so callers do not need to.

Reply:
If this logic does end up in the internal/config package, the runner will be much simpler. It would just be a set for set and a get for get. The verification of type and the callbacks would be handled in the lower-level package.

Reply:
This function isn't necessary anymore with some of the changes I made to the config rules